community/collections/src/main/java/org/neo4j/helpers/collection/Iterables.java

```diff
@@ -612,7 +628,10 @@ public static String toString( Iterable<?> values, String separator )
      */
     public static <T> long count( Iterable<T> iterable, Predicate<T> filter )
     {
-        return Iterators.count( iterable.iterator(), filter );
+        try ( ResourceIterator<T> iterator = Iterators.asResourceIterator( iterable.iterator() ) )
```

Comment:
Instead of wrapping in a resource iterator, which is potentially one unnecessary object instantiation, couldn't we just do the usual check of `Resource` in `finally`?